cmake/HPX_Libraries.cmake

```diff
@@ -25,11 +25,35 @@ macro(hpx_libraries)
     if(NOT _skip)
       list(FIND HPX_LIBRARIES "${lib}" _found)
       if(_found EQUAL -1)
-        set(_libs ${_libs} ${_keyword} ${lib})
+        if(IS_ABSOLUTE ${lib})
+          hpx_create_imported_library(${lib} _target)
+          set(_libs ${_libs} ${_keyword} ${_target})
+        else()
+          set(_libs ${_libs} ${_keyword} ${lib})
+        endif()
       endif()
       set(_keyword "")
       set(_skip FALSE)
     endif()
   endforeach()
   set(HPX_LIBRARIES ${HPX_LIBRARIES} ${_libs} CACHE INTERNAL "" FORCE)
 endmacro()
+
+macro(hpx_create_imported_library path target)
+  get_filename_component(_basename ${path} NAME_WE)
+  set(${target} hpx::imported::${_basename})
+
+  if(NOT TARGET ${${target}})
+    get_filename_component(_extension ${path} EXT)
+    if(_extension STREQUAL ".so")
+      set(_link_type SHARED)
+    elseif(_extension STREQUAL ".a")
+      set(_link_type STATIC)
+    else()
+      message(FATAL_ERROR "Unknown library extension (${_extension}): ${path}")
```

Comment:
Actually it does break with something akin to this:
```
CMake Error at cmake/HPX_SetupTarget.cmake:201 (target_link_libraries):
  Target "hpx::imported::libboost_atomic" of type UNKNOWN_LIBRARY may not be
  linked into another target.  One may link only to STATIC or SHARED
  libraries, or to executables with the ENABLE_EXPORTS property set.
Call Stack (most recent call first):
  cmake/HPX_AddLibrary.cmake:163 (hpx_setup_target)
  plugins/parcel/coalescing/CMakeLists.txt:11 (add_hpx_library)
  plugins/parcel/CMakeLists.txt:16 (add_coalescing_module)
  plugins/CMakeLists.txt:19 (add_parcel_modules)
  CMakeLists.txt:1652 (add_plugin_modules)
```
which is why I added the check in the first place. Further changes on my behalf probably will require a better understanding of the HPX build system...